provision/bin/install-postflight.sh

```diff
@@ -11,12 +11,27 @@ echo ">>> Installing DC/OS Postflight: /usr/local/sbin/dcos-postflight"
 
 cat << 'EOF' > "/usr/local/sbin/dcos-postflight"
 #!/usr/bin/env bash
+shopt -s nullglob
+
 # Run the DC/OS diagnostic script for up to the specified number of seconds to ensure
 # we do not return ERROR on a cluster that hasn't fully achieved quorum.
 TIMEOUT_SECONDS="${1:-900}"
-if [[ -e "/opt/mesosphere/bin/3dt" ]]; then
-    # DC/OS >= 1.7
+if [[ -e "/opt/mesosphere/bin/dcos-diagnostics" ]]; then
+    # DC/OS >= 1.10
+    CMD="/opt/mesosphere/bin/dcos-diagnostics --diag"
+elif [[ -e "/opt/mesosphere/bin/3dt" ]]; then
+    # DC/OS <= 1.9
+    # 3dt --diag will complain about missing endpoints_config.json if not
+    # passed explicitly. This error is not influencing the diagnostics status.
+    # This is a bug, which has been fixed in DC/OS >= 1.10
     CMD="/opt/mesosphere/bin/3dt --diag"
+    cfg_files=( /opt/mesosphere/etc/dcos-3dt-endpoint-config*.json )
+    if [ ${#cfg_files[@]} -gt 0 ]; then
+        CMD="${CMD} --endpoint-config=${cfg_files[0]}"
+    else
+        cfg_files=( /opt/mesosphere/packages/3dt*/endpoints_config.json )
```

Comment:
Can you do some spelunking and figure out when this file was moved so we can comment it here?
That way we'll know when it's safe to remove the workaround. We only keep compatibility back 2-3 DC/OS versions.